subprojects/docs/src/docs/userguide/jvm/building_java_projects.adoc

```diff
@@ -258,12 +263,18 @@ Defines the minimum JVM version your code should run on, i.e. it determines the
 
 These options can be set per link:{groovyDslPath}/org.gradle.api.tasks.compile.JavaCompile.html[JavaCompile] task, or on the `java { }` extension for all compile tasks, using properties with the same names.
 
-However these options do not protect against the use of APIs introduced in later Java versions.
+[NOTE]
+====
+Using a toolchain makes it illegal to configure the `sourceCompatibility` or `targetCompatibility` at the `java { }` extension level.
+====
+
+However, these options do not protect against the use of APIs introduced in later Java versions.
 
 ==== Compiling and testing Java 6/7
 
 Gradle can only run on Java version 8 or higher.
-Gradle still supports compiling, testing, generating Javadoc and executing applications for Java 6 and Java 7. Java 5 is not supported.
+Gradle still supports compiling, testing, generating Javadoc and executing applications for Java 6 and Java 7.
```

Comment:
Have we confirmed that Java 6 and 7 work with toolchains?

Reply:
Java 7 for sure. I do not think someone did an explicit test with 6.

Reply:
#14518 proves it works for Java 6 as well